tools/gulptasks/dist-upload.js

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) Highsoft AS
+ */
+
+/* eslint func-style: 0, no-console: 0, max-len: 0 */
+const gulp = require('gulp');
+const glob = require('glob');
+const build = require('./lib/build');
+const { uploadFiles, uploadProductPackage } = require('./unsorted/uploadS3');
+
+
+const DIST_DIR = 'build/dist';
+
+/**
+ * Upload a distribution version for all products.
+ *
+ * @param {string} productName from build properties, e.g 'highstock'.
+ * @param {object} productProps containing product name and version (build.properties).
+ * @return {Promise<any> | Promise | Promise} Promise to keep
+ */
+function uploadPackage(productName, productProps) {
+    const { name, version, cdnpath } = productProps;
+    return uploadProductPackage(productName, cdnpath, name, version);
+}
+
+
+const distUpload = products => {
```

Comment:
I favor real functions with own scope, because then call & order come not into play, as functions get always parsed before everything else.

Reply:
Fixed. I wasn't aware of this. Thanks :+1: